kratos/utilities/specifications_utilities.h

```diff
@@ -124,6 +124,16 @@ namespace SpecificationsUtilities
      */
     void KRATOS_API(KRATOS_CORE) AddMissingVariables(ModelPart& rModelPart);
 
+    /**
+     * @brief This method adds to the model part the missing variables
+     * @param rModelPart Reference to the ModelPart containing the problem
+     * @param rEntitiesList List of entities check specifications
+     */
+    void KRATOS_API(KRATOS_CORE) AddMissingVariablesFromEntitiesList(
+        ModelPart& rModelPart,
+        const Parameters& rEntitiesList
```

Comment:
```suggestion
        const Parameters EntitiesList
```

Reply:
I thought we fixed that issue

Reply:
What do you mean?
There is no issue

Reply:
I think we fixed when we updated to json library

Reply:
This was never an issue of the json lib but a delicate design choice on our side
The json object is stored with a shared ptr, hence the Parameters object should je passed by value to avoid dangling references and other annoying things like initialization of classes that have a Parameters ref

Reply:
Okay

Reply:
BTW, we use referenced parameters in several parts of the code, most of them mine, should I remove it?

Reply:
not sure, I guess it can be done when those parts are touched the next time 🤷‍♂️